src/hashset.c

```diff
@@ -1511,7 +1523,6 @@ unsigned hashsetSampleElements(hashset *t, void **dst, unsigned count) {
     while (samples.count < count) {
         cursor = hashsetScan(t, cursor, sampleElementsScanFn, &samples, HASHSET_SCAN_SINGLE_STEP);
     }
-    rehashStepOnReadIfNeeded(t);
```

Comment:
You're moving this back to scan? It's been moving back and forth, but not doing it in scan seems to be more similar to what dict does.

Reply:
just a mistake - I think I overwrote changes at some point when I shouldn't have 😅